apps/meteor/tests/end-to-end/api/livechat/contacts.ts

```diff
@@ -632,4 +632,95 @@ describe('LIVECHAT - contacts', () => {
+		});
+
+		it('should return only contacts that match the searchText using email', async () => {
+			const res = await request.get(api(`omnichannel/contacts.search`)).set(credentials).query({ searchText: contact.emails[0] });
+			expect(res.status).to.be.equal(200);
+			expect(res.body).to.have.property('success', true);
+			expect(res.body.contacts).to.be.an('array');
+			expect(res.body.contacts.length).to.be.equal(1);
+			expect(res.body.total).to.be.equal(1);
+			expect(res.body.contacts[0]._id).to.be.equal(contactId);
+			expect(res.body.contacts[0].name).to.be.equal(contact.name);
+			expect(res.body.contacts[0].emails[0]).to.be.equal(contact.emails[0]);
+		});
+
+		it('should return only contacts that match the searchText using phone number', async () => {
+			const res = await request.get(api(`omnichannel/contacts.search`)).set(credentials).query({ searchText: contact.phones[0] });
+			expect(res.status).to.be.equal(200);
+			expect(res.body).to.have.property('success', true);
+			expect(res.body.contacts).to.be.an('array');
+			expect(res.body.contacts.length).to.be.equal(1);
+			expect(res.body.total).to.be.equal(1);
+			expect(res.body.contacts[0]._id).to.be.equal(contactId);
+			expect(res.body.contacts[0].name).to.be.equal(contact.name);
+			expect(res.body.contacts[0].emails[0]).to.be.equal(contact.emails[0]);
+		});
+
+		it('should return only contacts that match the searchText using name', async () => {
+			const res = await request.get(api(`omnichannel/contacts.search`)).set(credentials).query({ searchText: contact.name });
+			expect(res.status).to.be.equal(200);
+			expect(res.body).to.have.property('success', true);
+			expect(res.body.contacts).to.be.an('array');
+			expect(res.body.contacts.length).to.be.equal(1);
+			expect(res.body.total).to.be.equal(1);
+			expect(res.body.contacts[0]._id).to.be.equal(contactId);
+			expect(res.body.contacts[0].name).to.be.equal(contact.name);
+			expect(res.body.contacts[0].emails[0]).to.be.equal(contact.emails[0]);
+		});
+
+		it('should return an empty list if no contacts exist', async () => {
+			const res = await request.get(api(`omnichannel/contacts.search`)).set(credentials).query({ searchText: 'invalid' });
+			expect(res.status).to.be.equal(200);
+			expect(res.body).to.have.property('success', true);
+			expect(res.body.contacts).to.be.an('array');
+			expect(res.body.contacts.length).to.be.equal(0);
+			expect(res.body.total).to.be.equal(0);
+		});
+
+		it("should return an error if user doesn't have 'view-livechat-contact' permission", async () => {
+			await removePermissionFromAllRoles('view-livechat-contact');
+
+			const res = await request.get(api(`omnichannel/contacts.search`)).set(credentials);
+
+			expect(res.body).to.have.property('success', false);
+			expect(res.body.error).to.be.equal('User does not have the permissions required for this action [error-unauthorized]');
+
+			await restorePermissionToRoles('view-livechat-contact');
```

Comment:
this is gonna happen on the `after` anyways 🤔 idk if it would be better to put this one on a separate describe block tho

Reply:
It's just to ensure that the permission will be restored. Every test needs to be done and undo everything required, right?

Reply:
yeah, i mean, on the describe block you already have this call. That means the "restore" will be executed 2 times. Not a big deal since it's on tests. 

So was thinking that this case could be in its own describe. But take this with a grain of salt as this is my personal pref.

Reply:
Make sense since we already does this for another test